bids/layout/index.py

```diff
@@ -31,12 +31,12 @@ def _check_path_matches_patterns(path, patterns):
     """Check if the path matches at least one of the provided patterns. """
     if not patterns:
         return False
-    path = os.path.abspath(path)
+    path = path.absolute()
     for patt in patterns:
-        if isinstance(patt, str):
+        if isinstance(patt, Path):
             if path == patt:
                 return True
```

Comment:
```suggestion
        if str(path) == str(patt):
                return True
```

Reply:
Simpler:

```suggestion
        return str(path) == patt
```

Reply:
`patterns` are either `self._include_patterns` or `self._exclude_patterns` when called from `_validate_*` functions. These are `None` by default and are populated in `self.__call__` where they are set to whatever `validation.validate_indexing_args` returns. That function only returns `Path` and `SRE_Pattern` objects so no pattern is ever of type str. So, we just need to check whether a given pattern is of type `Path`.

Reply:
But we *do* need to check for the type to differentiate `Path` from `SRE_Pattern`, so we can't just check for `str(path) == str(patt)`.

As for the simplification: we don't want to return if the comparison is False - we want to continue the loop, so returning should be only done when the comparison is True.

Reply:
In other words, this wouldn't work:

```suggestion
        if isinstance(patt, Path):
            if path == patt:
                return True
            else:
                return False
```